pkg/backoff/backoff_test.go

```diff
@@ -0,0 +1,67 @@
+package backoff
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestImmediate(t *testing.T) {
+	t.Parallel()
+
+	bkoff := New()
+	bkoff.delay = 0.0
```

Comment:
probably shouldn't be accessing private vars on the struct here (i.e. this file should have `package backoff_test`). 

That said, it's awesome that there are tests here.